caas/kubernetes/provider/configmap.go

```diff
@@ -18,9 +18,11 @@ import (
 )
 
 func (k *kubernetesClient) getConfigMapLabels(appName string) map[string]string {
-	return map[string]string{
-		constants.LabelApplication: appName,
+	labels := utils.LabelsForApp(appName, k.IsLegacyLabels())
+	if !k.IsLegacyLabels() {
+		labels = utils.LabelsMerge(labels, utils.LabelsJuju)
 	}
```

Comment:
No it can't as we have then putting a blanket statement on all users of labels for app. It will then trickle down into our selectors for subordinate objects.